### ISSUE TITLE
Site Editor: Content-width and wide widths sometimes don't apply when browsing through Pages

### ISSUE THREAD

**Issue by jasmussen** [WordPress/gutenberg#59593]
This one is hard to describe, so it's best to describe steps to reproduce:

* Have some pages saved/published on your site. 
* Have a site design, ideally a Header template part, that leverages the _wide_ width for the main containing group. E.g. Twenty TwentyFour by default.
* Go to the site editor (/wp-admin/site-editor.php).
* Navigate to "Pages"
* Select a page, and observe how the wide content-width for the header template becomes full-wide (or just unbounded).
* Now reload the page and observe how upon reload, the same wide content-width for the header template is correct.

The above steps are shown in these two GIF files as well:

![wide alignment in page editor](https://github.com/WordPress/gutenberg/assets/1204802/114afbc5-f9c4-4659-aa82-22f9e5177498)
![reloading to fix it](https://github.com/WordPress/gutenberg/assets/1204802/1bad5628-54c2-4996-a434-1451459f23a1)

In other words, the wide width is not respected when browsing to a page, that incorrect width even persists inside the editor. Until you reload in either the site editor page detail page, or the fullscreen editor. So some width-related CSS is missing in one case, but not the other.

**Comment by jasmussen** [comment]
Testing this more, here are the results:

* 6.4.3: not affected by this issue.
* 6.4.3 with the trunk Gutenberg plugin: can reproduce.
* 6.5 nightly: can reproduce. (With or without the Gutenberg trunk plugin.)

**Comment by andrewserong** [comment]
I'm not too sure the cause, but from a (very) quick test, it looks like the `.has-global-padding` rules aren't being output (potentially) when we go to view the individual page. However, upon reloading the site editor, these rules are applying correctly:

<img width="445" alt="image" src="https://github.com/WordPress/gutenberg/assets/14988353/f2d648d4-a945-4e6d-9e31-ddbc6e859468">

So in terms of next steps to investigate, I wonder why the global padding rules might be being skipped or not being output correctly in this case?

I might not get a chance to dig any deeper until later on in the week since I'm travelling.
